File: server-mock/src/main/java/com/belogrudov/servermock/model/Politician.java
package com.belogrudov.servermock.model;

import java.time.LocalDate;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Builder;
import lombok.Value;
import lombok.extern.jackson.Jacksonized;

@Value
@Builder
@Jacksonized
@JsonIgnoreProperties(ignoreUnknown = true)
public class Politician {
    String name;
    String surname;
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    LocalDate birthdate;
    @JsonProperty("country_code")
    String countryCode;
    String bio;
    List<Promise> promises;

    public String toJson() {
        return """
                {
                    "name":"%s",
                    "surname":"%s",
                    "birthdate":"%s",
                    "country_code":"%s",
                    "bio":"%s",
                    "promises":%s
                }"""
                .formatted(name, surname, birthdate, countryCode, bio, promises);
    }
}

File: frontend/lib/model/Politician.dart
import 'Promise.dart';

class Politician {
  final String name;
  final String surname;
  final String birthdate;
  final String countryCode;
  final String? bio;
  final List<Promise>? promises;

  const Politician({
    required this.name,
    required this.surname,
    required this.birthdate,
    required this.countryCode,
    required this.bio,
    required this.promises,
  });

  factory Politician.fromJson(Map<String, dynamic> json) {
    return Politician(
      name: json['name'],
      surname: json['surname'],
      birthdate: json['birthdate'],
      countryCode: json['country_code'],
      bio: json['bio'],
      promises: inferPromises(json["promises"]),
    );
  }

  static List<Promise>? inferPromises(List? jsonPromises) {
    return jsonPromises?.map((promise) => Promise.fromJson(promise)).toList();
  }

  @override
  String toString() {
    return '$name $surname, $birthdate, $countryCode';
  }
}
